src/rest/json.cpp

```diff
@@ -0,0 +1,308 @@
+/*
+ *  Copyright (c) 2017, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "rest/json.hpp"
+
+namespace otbr {
+namespace rest {
+
+static uint16_t HostSwap16(uint16_t v)
+
+{
+    return (((v & 0x00ffU) << 8) & 0xff00) | (((v & 0xff00U) >> 8) & 0x00ff);
+}
+
+std::string JSON::TwoVectorToJson(const std::vector<std::string> &aKey, const std::vector<std::string> &aValue)
+{
+    cJSON *           json = cJSON_CreateObject();
+    std::string       key;
+    cJSON *           value;
+    long unsigned int index;
+
+    for (index = 0; index < aKey.size(); index++)
+    {
+        key   = aKey[index];
+        value = cJSON_Parse(aValue[index].c_str());
+        cJSON_AddItemToObject(json, key.c_str(), value);
+    }
+    return JsonToStringDeleteJson(json);
+}
+
+std::string JSON::VectorToJson(const std::vector<std::string> &aVector)
+{
+    cJSON *json = cJSON_CreateArray();
+    cJSON *item;
+
+    for (auto str : aVector)
+    {
+        item = cJSON_Parse(str.c_str());
```

Comment:
If we already have a vector of valid JSON strings, why don't we just output `[`, then output each element separated by `, `, then output `]`.